lib/std/mem.zig

```diff
@@ -1899,22 +1899,38 @@ pub fn replace(comptime T: type, input: []const T, needle: []const T, replacemen
 test "replace" {
     var output: [29]u8 = undefined;
     var replacements = replace(u8, "All your base are belong to us", "base", "Zig", output[0..]);
+    var expected: []const u8 = "All your Zig are belong to us";
     testing.expect(replacements == 1);
-    testing.expect(eql(u8, output[0..], "All your Zig are belong to us"));
+    testing.expect(eql(u8, output[0..expected.len], expected));
 
     replacements = replace(u8, "Favor reading code over writing code.", "code", "", output[0..]);
+    expected = "Favor reading  over writing .";
     testing.expect(replacements == 2);
-    testing.expect(eql(u8, output[0..], "Favor reading  over writing ."));
+    testing.expect(eql(u8, output[0..expected.len], expected));
+
+    // Try adjacent replacements.
+
+    replacements = replace(u8, "\\n\\n", "\\n", "\n", output[0..]);
+    expected = "\n\n";
+    testing.expect(replacements == 2);
+    testing.expect(eql(u8, output[0..expected.len], expected));
+
+    replacements = replace(u8, "abbba", "b", "cd", output[0..]);
+    expected = "acdcdcda";
+    testing.expect(replacements == 3);
+    testing.expect(eql(u8, output[0..expected.len], expected));
 }
 
 /// Calculate the size needed in an output buffer to perform a replacement.
 pub fn replacementSize(comptime T: type, input: []const T, needle: []const T, replacement: []const T) usize {
     var i: usize = 0;
     var size: usize = input.len;
-    while (i < input.len) : (i += 1) {
+    while (i < input.len) {
         if (mem.indexOf(T, input[i..], needle) == @as(usize, 0)) {
             size = size - needle.len + replacement.len;
             i += needle.len;
+        } else {
```

Comment:
`std.mem.replacementSize(u8, "foo", "", "a")` will now enter an endless loop.
Passing an empty string as needle is somewhat useful to intersperse the replacement between each item, we may want to handle it or filter it out with an assertion.

Reply:
Rust, Go, JS, etc support this, and I think Zig should too. It currently doesn't on master (at least `replaceOwned` panics if given an empty needle), so maybe that could be a follow-up PR and just assert in this one.

Reply:
I prefer to add an assert for this for now, and implement "intersperse" as a separate PR. I tried this (empty needle) with mem.replace and it just loops until the output buffer overflows. So we currently don't support this in any form. I'll add the assert to all the appropriate places and retest.

Reply:
I think that plan makes sense 👍